client/components/sites-dropdown/test/index.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
+import useFilesystemMocks from 'test/helpers/use-filesystem-mocks';
+
+describe( 'index', function() {
+	useFakeDom();
+
+	useFilesystemMocks( __dirname );
+
+	useMockery( mockery => {
+		mockery.registerSubstitute( 'matches-selector', 'component-matches-selector' );
+	} );
+
+	let SitesDropdown;
+
+	before( function() {
+		SitesDropdown = require( '../index.jsx' );
+	} );
+
+	describe( 'component rendering', function() {
+		it( 'should render a dropdown component initially closed', function() {
+			const sitesDropdown = shallow( <SitesDropdown /> );
+			expect( sitesDropdown.hasClass( 'sites-dropdown' ) ).to.be.true;
+			expect( sitesDropdown.hasClass( 'is-open' ) ).to.be.false;
+		} );
+
+		it( 'should toggle the dropdown, when it is clicked', function() {
+			const toggleOpenSpy = sinon.spy( SitesDropdown.prototype, 'toggleOpen' );
+			const sitesDropdown = shallow( <SitesDropdown /> );
+
+			sitesDropdown.find( '.sites-dropdown__selected' ).simulate( 'click' );
+
+			sinon.assert.calledOnce( toggleOpenSpy );
+			expect( sitesDropdown.hasClass( 'is-open' ) ).to.be.true;
+
+			toggleOpenSpy.restore();
+		} );
+	} );
+
+	describe( 'component state', function() {
+		it( "should initially consider as selected the user's primary site, when is not specified something different", function() {
+			const sitesDropdown = shallow( <SitesDropdown /> );
+			expect( sitesDropdown.instance().state.selectedSiteSlug ).to.be.eql( 'primary.wordpress.com' );
+		} );
+
+		it( 'should initially consider as selected the site whose id is passed as `selectedSiteId` prop', function() {
+			const sitesDropdown = shallow( <SitesDropdown selectedSiteId={ 42 } /> );
+			expect( sitesDropdown.instance().state.selectedSiteSlug ).to.be.eql( 'foo.wordpress.com' );
+		} );
+	} );
+
+	describe( 'selectSite', function() {
+		it( 'should update the `selectedSiteSlug`, and `open` state properties', function() {
+			const fakeContext = prepareFakeContext();
+			const { setState: setStateSpy, props: { onSiteSelect: siteSelectedSpy } } = fakeContext;
```

Comment:
I think it's clearer to define the spies and then construct the context object. With the destructure here I need to look up the object shape below.

For tests it's OK to repeat yourself. I like keeping setup as close as possible to the test, so we can read a block and understand it without having to jump around as much in the file.